docs/api/en/core/Object3D.html

```diff
@@ -438,11 +438,22 @@ <h3>[method:null traverseAncestors]( [param:Function callback] )</h3>
 		</p>
 
 		<h3>[method:null updateMatrix]()</h3>
-		<p>Update the local transform.</p>
+		<p>Updates the local transform matrix, signals to update global transform matrix before rendering.</p>
 
 		<h3>[method:null updateMatrixWorld]( [param:Boolean force] )</h3>
-		<p>Update the global transform of the object and its children.</p>
+		<p>
+		force - immediately updates global transform of the object and its children.<br /><br />
```

Comment:
I'm not sure this description is correct. The `force` parameter just ensures to update the world matrix if `updateMatrix()` was not called. This is the case if `matrixAutoUpdate` is set to `false` and the user did not manually use `updateMatrix()`.

Reply:
Hmm, you're right. But it's basically the same meaning, cause with `force` it does immediately update the global transforms. What would be the best way to phrase it then?

Also, is there any scenarios where you would call `updateMatrixWorld` without `force = true`?
Cause the world matrices aren't even guaranteed to update without it.

Reply:
This function is actually quite misleading. It's hard to tell if the main purpose is updating the local transform matrix or the global transform. The general conditional execution is also very weird. I know @WestLangley suggested a different approach for it a while ago, but I'm not sure how it stands currently.

Reply:
>  It's hard to tell if the main purpose is updating the local transform

Definitely not. If you just want to update the local transformation matrix, use `Object3D.updateMatrix()`.

Reply:
> Also, is there any scenario where you would call updateMatrixWorld without force = true?

Normally, users don't have to worry about the `force` flag. You should update the transformation of 3D objects via `Object3D.position`, `Object3D.rotation`/`Object3D.quaternion` and `Object3D.scale`. If you need for some reason the local matrix, call `Object3D.updateMatrix()`. If you need the world matrix, call `Object3D.updateMatrixWorld()` (without `force`).

The big issue with `Object3D.updateMatrixWorld()` is that it does not respect an update of its ancestors. It only updates the current object (itself) and its descendants. `Object3D.updateWorldMatrix()` should solve this issue.

AIFAIK, `force` is only relevant for two use cases: 

- If the user sets `matrixAutoUpdate` to `false`(meaning he decides when to update the local matrix) and directly assigns a local matrix (meaning he does not use `Object3D.updateMatrix()`). In this case `Object3D.matrixWorldNeedsUpdate` will always be `false` and thus `force` is required.
- To force an update of ancestors. That is necessary if the current object transformation has changed. But this is more an internal thing and not relevant to users.

Reply:
In any event, the PR in its current form is not an improvement. Without improvements, I suggest to close it and leave the documentation as it is. The following guide also explains the handling with matrices.

https://threejs.org/docs/index.html#manual/en/introduction/Matrix-transformations

Reply:
I'll be pushing an update shortly, if it's still not considered an improvement then we can close this PR and move on.